### ISSUE TITLE
Internal server error 500 Server in trouble

### ISSUE THREAD

**Issue by hugokoopmans** [versae/ipython-cypher#3]
Hi there,

Thx for your great work so far... looking forward to get all of it to work.
I installed Neo4J default, python default ipython notebook default but cannot get things to run...
I am on linux mint 17
python 2.7

notebook

    In [1] : %load_ext cypher

    In [2]: %config CypherMagic

CypherMagic options
###-----------------
CypherMagic.auto_html=<Bool>
    Current: False
    Return a D3 representation of the graph instead of regular result sets
CypherMagic.auto_limit=<Int>
    Current: 0
    Automatically limit the size of the returned result sets
CypherMagic.auto_networkx=<Bool>
    Current: False
    Return Networkx MultiDiGraph instead of regular result sets
CypherMagic.auto_pandas=<Bool>
    Current: False
    Return Pandas DataFrame instead of regular result sets
CypherMagic.data_contents=<Bool>
    Current: True
    Bring extra data to render the results as a graph
CypherMagic.display_limit=<Int>
    Current: 0
    Automatically limit the number of rows displayed (full result set is still
    stored)
CypherMagic.feedback=<Bool>
    Current: True
    Print number of rows affected
CypherMagic.rest=<Bool>
    Current: False
    Return full REST representations of objects inside the result sets
CypherMagic.short_errors=<Bool>
    Current: True
    Don't display the full traceback on Neo4j errors
CypherMagic.style=<Unicode>
    Current: u'DEFAULT'
    Set the table printing style to any of prettytable's defined styles
    (currently DEFAULT, MSWORD_FRIENDLY, PLAIN_COLUMNS, RANDOM)

basically as soon as i try to do someting in ipython i get the error 500 error

So

    In [3]: %%cypher http://localhost:7474/db/data
    match (n) return n limit 1

I get error:

    Code [500]: Internal Server Error. Server got itself in trouble.
    Element not found

/usr/lib/pymodules/python2.7/matplotlib/__init__.py:817: UserWarning: svg.embed_char_paths is deprecated and replaced with svg.fonttype; please use the latter.
  warnings.warn(self.msg_depr % (key, alt))
/usr/lib/pymodules/python2.7/matplotlib/__init__.py:817: UserWarning: savefig.extension is deprecated and replaced with savefig.format; please use the latter.
  warnings.warn(self.msg_depr % (key, alt))

any suggestions?

**Comment by versae** [comment]
@hugokoopmans could you please let me know what versions you have of Neo4j and IPython? New Neo4j 2.2 now force to change the password the first time, and that is mechanism that is not implemented yet in `ipython-cypher`. So, could you pleas try disabling auth in Neo4j first?

**Comment by hugokoopmans** [comment]
@versae Sure:
iPython
Python 2.7.6 (default, Mar 22 2014, 22:59:56) 
Type "copyright", "credits" or "license" for more information.

IPython 1.2.1 -- An enhanced Interactive Python.

And neo4j is neo4j-community-2.1.7

    :GET /db/manage/server/jmx/domain/org.neo4j/instance%3Dkernel%230%2Cname%3DConfiguration

[{"description":"The configuration parameters used to configure Neo4j","name":"org.neo4j:instance=kernel#0,name=Configuration","attributes":[{"description":"Configuration attribute","name":"store_dir","value":"/home/hugo/tools/neo4j-community-2.1.7/data/graph.db","isReadable":"true","type":"java.lang.String","isWriteable":"false ","isIs":"false "},{"description":"Configuration attribute","name":"neostore.nodestore.db.mapped_memory","value":"148M","isReadable":"true","type":"java.lang.String","isWriteable":"false ","isIs":"false "},{"description":"Enable a remote shell server which shell clients can log in to","name":"remote_shell_enabled","value":"true","isReadable":"true","type":"java.lang.String","isWriteable":"false ","isIs":"false "},{"description":"Configuration attribute","name":"neostore.relationshipgroupstore.db.mapped_memory","value":"0M","isReadable":"true","type":"java.lang.String","isWriteable":"false ","isIs":"false "},{"description":"Configuration attribute","name":"neostore.propertystore.db.strings.mapped_memory","value":"470M","isReadable":"true","type":"java.lang.String","isWriteable":"false ","isIs":"false "},{"description":"Configuration attribute","name":"neostore.propertystore.db.arrays.mapped_memory","value":"548M","isReadable":"true","type":"java.lang.String","isWriteable":"false ","isIs":"false "},{"description":"Configuration attribute","name":"neostore.relationshipstore.db.mapped_memory","value":"635M","isReadable":"true","type":"java.lang.String","isWriteable":"false ","isIs":"false "},{"description":"Configuration attribute","name":"neostore.propertystore.db.mapped_memory","value":"537M","isReadable":"true","type":"java.lang.String","isWriteable":"false ","isIs":"false "}],"url":"org.neo4j/instance%3Dkernel%230%2Cname%3DConfiguration"}]

**Comment by versae** [comment]
Hi @hugokoopmans, I tried with the same versions, you can [take a look here](http://nbviewer.ipython.org/gist/versae/d5de145259bf50d951e8). But still haven't been able to reproduce it. I don't have Linux Mint but I will try during this weekend.

Also try this, so we can see if it's a `ipython-cypher` problem, or `neo4jrestclient` or something else.

**Comment by versae** [comment]
Sorry, what I meant was this
```python
from neo4jrestclient.client import GraphDatabase
gdb = GraphDatabase("http://localhost:7474/db/data")
gdb.query("match (n) return n limit 1")[:]
```
Did it work?

**Comment by hugokoopmans** [comment]
Hi @versae , well this gives:

In [1]:

```
from neo4jrestclient.client import GraphDatabase
gdb = GraphDatabase("http://localhost:7474/db/data")
gdb.query("match (n) return n limit 1")[:]
```

Out[1]:

```
[[{u'all_relationships': u'http://localhost:7474/db/data/transaction/commitdb/data/node/0/relationships/all',
   u'all_typed_relationships': u'http://localhost:7474/db/data/transaction/commitdb/data/node/0/relationships/all/{-list|&|types}',
   u'create_relationship': u'http://localhost:7474/db/data/transaction/commitdb/data/node/0/relationships',
   u'data': {u'released': 1999,
    u'tagline': u'Welcome to the Real World',
    u'title': u'The Matrix'},
   u'incoming_relationships': u'http://localhost:7474/db/data/transaction/commitdb/data/node/0/relationships/in',
   u'incoming_typed_relationships': u'http://localhost:7474/db/data/transaction/commitdb/data/node/0/relationships/in/{-list|&|types}',
   u'labels': u'http://localhost:7474/db/data/transaction/commitdb/data/node/0/labels',
   u'metadata': {u'id': 0, u'labels': [u'Movie']},
   u'outgoing_relationships': u'http://localhost:7474/db/data/transaction/commitdb/data/node/0/relationships/out',
   u'outgoing_typed_relationships': u'http://localhost:7474/db/data/transaction/commitdb/data/node/0/relationships/out/{-list|&|types}',
   u'paged_traverse': u'http://localhost:7474/db/data/transaction/commitdb/data/node/0/paged/traverse/{returnType}{?pageSize,leaseTime}',
   u'properties': u'http://localhost:7474/db/data/transaction/commitdb/data/node/0/properties',
   u'property': u'http://localhost:7474/db/data/transaction/commitdb/data/node/0/properties/{key}',
   u'self': u'http://localhost:7474/db/data/transaction/commitdb/data/node/0',
   u'traverse': u'http://localhost:7474/db/data/transaction/commitdb/data/node/0/traverse/{returnType}'}]]
```

so this proves the rest client is ok?

hugo

**Comment by hugokoopmans** [comment]
si versions do match , only difference is OS itself:

In [2]:
```
! cat /etc/issue

print "Neo4j", gdb.VERSION

! python --version

! pip list | grep ipython

! pip list | grep neo4jrestclient
```

Linux Mint 17 Qiana \n \l
Neo4j 2.1.7
Python 2.7.6
ipython (1.2.1)
ipython-cypher (0.2.1)
neo4jrestclient (2.1.0)

**Comment by versae** [comment]
Hey! Did you ever found out the error?

**Comment by versae** [comment]
Closing the issue due to inactivity. Feel free to reopen.
